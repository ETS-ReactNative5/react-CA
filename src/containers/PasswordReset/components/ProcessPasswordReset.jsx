import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Aux from 'components/Hoc/Aux/Aux'
import Textbox from 'components/Core/Textbox/Textbox'
import Button from 'components/Core/Button/Button'
import { Row, Col } from 'components/Core/Section/Section'

import lockIcon from 'assets/images/dashboard/lockIcon.svg'
import arrowRight from 'assets/images/global/arrow-right.svg'
import styles from '../PasswordReset.scss'


class ProcessPasswordReset extends Component {
  state = {
    password: '',
    verifiedPassword: '',
  }

  handleSetPassword = password => this.setState(ps => ({ ...ps, password }))
  handleSetVerifiedPassword = verifiedPassword => this.setState(ps => ({ ...ps, verifiedPassword }))

  render() {
    const { password, verifiedPassword } = this.state
    const { loading, onSubmit } = this.props

    const minPasswordLength = 8
    const disabled = loading
      || (password !== verifiedPassword)
      || (password.length === 0 && verifiedPassword.length === 0)
      || (password.length < minPasswordLength && verifiedPassword.length < minPasswordLength)

    const errorMessage = verifiedPassword.length !== 0 && password !== verifiedPassword ? 'Your passwords don\'t match' : null
    const passwordValidator = [{ minLength: minPasswordLength, error: { level: 3, message: 'Your password must be at least 8 characters long!' } }]

    return (
      <Aux>
        <Row>
          <Col md='2' />
          <Col md='8'>
            <h1 className={styles.header}>Password Reset</h1>
            <p>
              You are only a step away from re-accessing your account,
                please create a password below to login.
            </p>
          </Col>
          <Col md='2' />
        </Row>
        <Row>
          <Col md='2' />
          <Col md='8'>
            <Textbox
              placeholder='Password'
              password
              icon={lockIcon}
              isTransparent
              handleValueChange={this.handleSetPassword}
              validators={passwordValidator}
            />
            {
              password.length > 0 && (<Textbox
                placeholder='Verify Password'
                password
                icon={lockIcon}
                isTransparent
                handleValueChange={this.handleSetVerifiedPassword}
                incomingError={errorMessage}
              />)
            }
          </Col>
          <Col md='2' />
        </Row>
        <Row>
          <div className={styles.buttonContainer}>
            <Button
              onClick={() => onSubmit(this.state.password)}
              disabled={disabled}
              icon={arrowRight}
              text='Reset Password'
              primary
            />
          </div>
        </Row>
      </Aux>
    )
  }
}

ProcessPasswordReset.propTypes = {
  loading: PropTypes.bool.isRequired,
  onSubmit: PropTypes.func.isRequired,
}


export default ProcessPasswordReset
