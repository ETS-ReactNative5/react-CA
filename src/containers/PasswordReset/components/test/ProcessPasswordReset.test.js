import React from 'react'
import { shallow } from 'enzyme'

import ProcessPasswordReset from '../ProcessPasswordReset'

const componentProps = {
  onSubmit: jest.fn(),
  loading: false,
}

describe('<ProcessPasswordReset />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ProcessPasswordReset {...componentProps} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('does not allow submission when loading prop is true', () => {
    const testProps = { ...componentProps, loading: true }
    const wrapper = shallow(<ProcessPasswordReset {...testProps} />)
    const button = wrapper.find('Button')
    expect(button).toBeDisabled()
  })

  it('does not allow submission when passwords do not match', () => {
    const wrapper = shallow(<ProcessPasswordReset {...componentProps} />)
    const button = wrapper.find('Button')

    wrapper.setState({ password: 'TobyMaguire', verifiedPassword: 'KirstenDunst' })
    wrapper.update()

    expect(button).toBeDisabled()
  })

  it('sends a valid password to the onSubmit callback', () => {
    const wrapper = shallow(<ProcessPasswordReset {...componentProps} />)
    const button = wrapper.find('Button')

    wrapper.setState({ password: 'TobyMaguireIsSpidey', verifiedPassword: 'TobyMaguireIsSpidey' })
    wrapper.update()
    button.simulate('click')

    expect(componentProps.onSubmit).toHaveBeenCalledWith('TobyMaguireIsSpidey')
  })
})
