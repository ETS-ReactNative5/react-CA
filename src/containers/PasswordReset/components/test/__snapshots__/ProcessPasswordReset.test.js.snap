// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProcessPasswordReset /> renders correctly 1`] = `
<Aux>
  <Row
    className=""
  >
    <Col
      className=""
      lg={null}
      lg-offset={null}
      md="2"
      md-offset={null}
      sm={null}
      sm-offset={null}
      xl={null}
      xl-offset={null}
      xs={null}
      xs-offset={null}
    />
    <Col
      className=""
      lg={null}
      lg-offset={null}
      md="8"
      md-offset={null}
      sm={null}
      sm-offset={null}
      xl={null}
      xl-offset={null}
      xs={null}
      xs-offset={null}
    >
      <h1
        className="header"
      >
        Password Reset
      </h1>
      <p>
        You are only a step away from re-accessing your account, please create a password below to login.
      </p>
    </Col>
    <Col
      className=""
      lg={null}
      lg-offset={null}
      md="2"
      md-offset={null}
      sm={null}
      sm-offset={null}
      xl={null}
      xl-offset={null}
      xs={null}
      xs-offset={null}
    />
  </Row>
  <Row
    className=""
  >
    <Col
      className=""
      lg={null}
      lg-offset={null}
      md="2"
      md-offset={null}
      sm={null}
      sm-offset={null}
      xl={null}
      xl-offset={null}
      xs={null}
      xs-offset={null}
    />
    <Col
      className=""
      lg={null}
      lg-offset={null}
      md="8"
      md-offset={null}
      sm={null}
      sm-offset={null}
      xl={null}
      xl-offset={null}
      xs={null}
      xs-offset={null}
    >
      <Textbox
        buttonAction={null}
        buttonIcon={null}
        buttonText={null}
        className=""
        defaultValue={null}
        disabled={false}
        handleCleanState={[Function]}
        handleValueChange={[Function]}
        icon="lockIcon.svg"
        iconOnly={false}
        incomingError=""
        isAddress={false}
        isButtonEnable={null}
        isProfile={false}
        isTransparent={true}
        isValid={null}
        mask={null}
        onBlur={null}
        onFocus={null}
        optional={false}
        password={true}
        placeholder="Password"
        textboxName={null}
        validators={
          Array [
            Object {
              "error": Object {
                "level": 3,
                "message": "Your password must be at least 8 characters long!",
              },
              "minLength": 8,
            },
          ]
        }
      />
    </Col>
    <Col
      className=""
      lg={null}
      lg-offset={null}
      md="2"
      md-offset={null}
      sm={null}
      sm-offset={null}
      xl={null}
      xl-offset={null}
      xs={null}
      xs-offset={null}
    />
  </Row>
  <Row
    className=""
  >
    <div
      className="buttonContainer"
    >
      <Button
        className=""
        disabled={true}
        icon="arrow-right.svg"
        iconOnly={false}
        middle={true}
        nocaps={false}
        onClick={[Function]}
        outline={false}
        primary={true}
        secondary={false}
        small={false}
        text="Reset Password"
        textOnly={false}
        tiny={false}
      />
    </div>
  </Row>
</Aux>
`;
